prov/sockets/src/sock_ep_msg.c

```diff
@@ -1100,128 +873,150 @@ static struct fi_info *sock_ep_msg_process_info(struct sock_conn_req *req)
 			    req->info.dest_addr, req->info.src_addr);
 }
 
+static void *sock_pep_req_handler(void *data)
+{
+	int ret, entry_sz;
+	struct sock_conn_req *conn_req;
+	struct fi_eq_cm_entry *cm_entry = NULL;
+	struct sock_conn_req_handle *handle = data;
+
+	conn_req = calloc(1, sizeof(*conn_req) + SOCK_EP_MAX_CM_DATA_SZ);
+	if (!conn_req) {
+		SOCK_LOG_ERROR("cannot allocate memory\n");
+		goto err;
+	}
+
+	ret = sock_cm_recv(handle->sock_fd, conn_req, sizeof(*conn_req));
+	if (ret) {
+		SOCK_LOG_ERROR("IO failed\n");
+		goto err;
+	}
+
+	sock_ep_cm_process_hdr(&conn_req->hdr);
+	if (conn_req->hdr.cm_data_sz) {
+		ret = sock_cm_recv(handle->sock_fd, conn_req->cm_data,
+				   conn_req->hdr.cm_data_sz);
+		if (ret) {
+			SOCK_LOG_ERROR("IO failed for cm-data\n");
+			goto err;
+		}
+	}
+
+	cm_entry = calloc(1, sizeof(*cm_entry) + conn_req->hdr.cm_data_sz);
+	if (!cm_entry) {
+		SOCK_LOG_ERROR("cannot allocate memory\n");
+		goto err;
+	}
+
+	handle->handle.fclass = FI_CLASS_CONNREQ;
+	handle->req = conn_req;
```

Comment:
conn_req will be lost in an error case, since handle->req is not set until here.